### PR TITLE
Time related fixes.

### PR DESCRIPTION
**Pull request by robholland**
Use Google::Protobuf::Timestamp methods for time conversion.

Remove _seconds suffixes from some attributes.

### CODE REVIEW

**Review by antstorm — COMMENTED** [review]
Looks good, needs rebase

**Comment by robholland** [comment]
Updated.

File: lib/temporal/activity/task_processor.rb [hunk 1]
```diff
@@ -48,7 +48,9 @@ def process
       attr_reader :task, :domain, :task_token, :activity_name, :activity_class, :client, :middleware_chain
 
       def queue_time_ms
-        ((task.started_timestamp - task.scheduled_timestamp_of_this_attempt) / 1_000_000).round
+        scheduled = task.current_attempt_scheduled_time.to_f
+        started = task.started_timestamp.to_f
+        ((started - scheduled) * 1_000).round
```
Comment (antstorm) [inline lib/temporal/activity/task_processor.rb:53]: There should be exactly the same logic in `lib/temporal/workflow/decision_task_processor.rb`, can you please apply these changes to it as well?

File: lib/temporal/workflow/execution_info.rb [hunk 2]
```diff
@@ -26,8 +26,8 @@ def self.generate_from(response)
           workflow: response.type.name,
           workflow_id: response.execution.workflowId,
           run_id: response.execution.run_id,
-          start_time: Utils.time_from_nanos(response.startTime),
-          close_time: Utils.time_from_nanos(response.closeTime),
+          start_time: response.start_time.to_time,
```
Comment (antstorm) [inline lib/temporal/workflow/execution_info.rb:29]: As far as I understand `response.start_time` is an instance of `Google::Protobuf::Timestamp`, which does not implement `#to_time`

**Review by antstorm — CHANGES_REQUESTED** [review]

File: lib/temporal/workflow/execution_info.rb (hunk 2 above)
Comment (robholland) [reply lib/temporal/workflow/execution_info.rb:29]: It's defined here I believe: https://github.com/protocolbuffers/protobuf/blob/master/ruby/lib/google/protobuf/well_known_types.rb#L74

Comment (antstorm) [reply lib/temporal/workflow/execution_info.rb:29]:
```ruby
>> timestamp = Google::Protobuf::Timestamp.new
>> timestamp.to_time

NoMethodError (undefined method `to_time' for <Google::Protobuf::Timestamp: seconds: 0, nanos: 0>:Google::Protobuf::Timestamp)
```

Comment (robholland) [reply lib/temporal/workflow/execution_info.rb:29]: Hmm. Will dig into that to understand why that's not present.

Comment (antstorm) [reply lib/temporal/workflow/execution_info.rb:29]: Had to explicitly `require 'google/protobuf/well_known_types'` to make it work I don't think it is loaded by default

Comment (robholland) [reply lib/temporal/workflow/execution_info.rb:27]: Yep, found an issue discussing that. Updated my PR to require it.

File: lib/temporal/activity/task_processor.rb (hunk 1 above)
Comment (robholland) [reply lib/temporal/activity/task_processor.rb:53]: Done.

**Review by antstorm — APPROVED** [review]
👍 

Let's also start adding up the specs where these are missing. Will save us from these small mistakes moving forward

File: lib/temporal/workflow/serializer/schedule_activity.rb [hunk 3]
```diff
@@ -43,7 +43,6 @@ def serialize_retry_policy(retry_policy)
             maximum_interval: retry_policy.max_interval,
             maximum_attempts: retry_policy.max_attempts,
             non_retriable_error_reasons: non_retriable_errors,
-            expiration_interval: retry_policy.expiration_interval
```
Comment (antstorm) [inline lib/temporal/workflow/serializer/schedule_activity.rb:46]: Hmm, this change is totally unrelated and questionable (https://temporalio.slack.com/archives/CTRCR8RBP/p1603714682311600), can we leave it out of this PR?

Comment (robholland) [reply lib/temporal/workflow/serializer/schedule_activity.rb:46]: The field no longer exists, I can split it into another PR if you'd prefer but it needs to be removed.

Comment (robholland) [reply lib/temporal/workflow/serializer/schedule_activity.rb:46]: Removed from this PR.

**Review by antstorm — APPROVED** [review]
